File: PY/Bryann Valderrama/Matemática/ax+by=n.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def solution(a, b, n):
    i = 0
    while i * a <= n:
        if (n - (i * a)) % b == 0:
            wr(f'x = {i}, y = {(n-(i*a))//b}')
            return
        i += 1
    wr(f'Sin solucion')


a = 2
b = 3
n = 7
solution(a, b, n)

File: C++/Bryann Valderrama/Matemática/Ax+By=n.cpp
#include <cstdlib>
#include <bits/stdc++.h>
using namespace std;

void solution(int a, int b, int n)
{
    for (int i = 0; i * a <= n; i++)
    {
        if ((n - (i * a)) % b == 0)
        {
            cout << "x = " << i << ", y = " << (n - (i * a)) / b << endl;
            return;
        }
    }
    cout << "Sin solucion" << endl;
}

int main(int argc, char const *argv[])
{
    int a = 2;
    int b = 3;
    int n = 10;
    solution(a, b, n);
    return 0;
}
